### PR TITLE
Add MetaMask "HD Key Tree" derivation paths

### PR DESCRIPTION
**Pull request by webmaster128**
This together with https://github.com/iov-one/iov-core/issues/404 allows us to generate MetaMask addresses from a mnemonic

### CODE REVIEW

**Comment by webmaster128** [comment]
@LucasIsasmendi this one is easier to review commit by commit

File: packages/iov-keycontrol/src/hdpaths.ts [hunk 1]
```diff
@@ -10,4 +10,37 @@ export class HdPaths {
...
+      Slip10RawIndex.hardened(bip44Purpose),
+      Slip10RawIndex.hardened(coinType),
+      Slip10RawIndex.hardened(account),
+      Slip10RawIndex.normal(change),
+      Slip10RawIndex.normal(address),
+    ];
+  }
+
+  /**
+   * The default MetaMask derivation path
```
Comment (ethanfrey) [inline packages/iov-keycontrol/src/hdpaths.ts]: :+1:

File: packages/iov-keycontrol/src/hdpaths.ts [hunk 2]
```diff
@@ -10,4 +10,37 @@ export class HdPaths {
     const iovPurpose = 4804438;
     return [Slip10RawIndex.hardened(iovPurpose), Slip10RawIndex.hardened(index)];
   }
+
+  /**
+   * Only use this for the real BIP-0044 with
```
Comment (ethanfrey) [inline packages/iov-keycontrol/src/hdpaths.ts]: Like this.

Can we add a slip10hardened() function or something similar to generate ed25519 addresses?
I agree with a different name to differentiate them, but a similar helper for bns/lisk would be great.

Comment (webmaster128) [reply packages/iov-keycontrol/src/hdpaths.ts]: slip10 does not specify any path sheme. What we want is

```
trezorStyleBip44LikeAccountBasedEd25519(coinType: number, a: number)
```

which covers Stellar, Lisk, NEM, Monero as documented here https://github.com/trezor/trezor-core/tree/master/docs/coins as well as RISE and maybe others.

Anyway, we need to be much more specific in what kind of paths we need. For BNS no final path scheme other than simple address was specified.

**Pull request merged**